### PR TITLE
Add black

### PR DESCRIPTION
**Pull request by kclowes**
**This is branched off of #40 so only the last commit needs to be reviewed here, and this should be merged after #40**

## What was wrong?

Adding black to the repo. This repo is small enough that I felt okay doing it all in one fell swoop.

## How was it fixed?

Added the black dependency, and added it to the lint commands.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/eth-hash/blob/master/newsfragments/README.md)

[//]: # (See: https://eth-hash.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-hash/blob/master/newsfragments/README.md)

- [x] Add .git-blame-ignore-revs file with commit number once it's finalized

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,9 +17,9 @@ multi_line_output=3
 use_parentheses=True
 
 [flake8]
-max-line-length= 100
+max-line-length=88
 exclude= venv*,.tox,docs,build
-ignore=
+extend-ignore=W503
```
Comment (fselmo) [inline tox.ini:22]: Just curious why the `ignore` -> `extend-ignore`. Seems like `ignore` is more appropriate? 🤔

**Review by fselmo — APPROVED** [review]
lgtm, just had a minor question on semantics 👍

File: tox.ini (hunk 1 above)
Comment (kclowes) [reply tox.ini:22]: I just got it from the black docs. The flake8 docs say:

> The difference to the [--ignore](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-ignore) option is, that this option can be used to selectively add individual codes without overriding the default list entirely.

I honestly can't decipher what that actually means, but it seems like it's less heavy-handed?

Comment (fselmo) [reply tox.ini:22]: Ah interesting... glad I asked. We should maybe change this in other libraries where we've implemented `black` as well just for consistency. Just saw it in their docs too  👍

Comment (kclowes) [reply tox.ini:22]: yeah, good idea!

**Review by pacrob — APPROVED** [review]
lgtm!

**Pull request merged**